### ISSUE TITLE
Spring-boot 2.6.x - circular references - history - JLineShellAutoConfiguration

### ISSUE THREAD

**Issue by jardelnovaes** [spring-projects/spring-shell#376]
After updating to spring-boot 2.6.3 (actually any 2.6.X) the Spring-shell app crashed.
It works fine using spring-boot 2.5.9.
It also works with `spring.main.allow-circular-references=true` and spring-boot 2.6.x

```
***************************
APPLICATION FAILED TO START
***************************

Description:

The dependencies of some of the beans in the application context form a cycle:

┌──->──┐
|  org.springframework.shell.jline.JLineShellAutoConfiguration (field private org.springframework.shell.jline.PromptProvider org.springframework.shell.jline.JLineShellAutoConfiguration.promptProvider)
└──<-──┘


Action:

Relying upon circular references is discouraged and they are prohibited by default. Update your application to remove the dependency cycle between beans. As a last resort, it may be possible to break the cycle automatically by setting spring.main.allow-circular-references to true.

2022-02-21 17:42:06.836  WARN 508927 --- [           main] s.c.a.AnnotationConfigApplicationContext : Exception thrown from ApplicationListener handling ContextClosedEvent

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'history' defined in class path resource [org/springframework/shell/jline/JLineShellAutoConfiguration$HistoryConfiguration.class]: Unsatisfied dependency expressed through method 'history' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'org.springframework.shell.jline.JLineShellAutoConfiguration': Unsatisfied dependency expressed through field 'promptProvider'; nested exception is org.springframework.beans.factory.BeanCurrentlyInCreationException: Error creating bean with name 'org.springframework.shell.jline.JLineShellAutoConfiguration': Requested bean is currently in creation: Is there an unresolvable circular reference?
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:800) ~[spring-beans-5.3.15.jar:5.3.15]
```

**Comment by jardelnovaes** [comment]
I created a simple app to test that behavior.
https://github.com/jardelnovaes/shell-test-issue-376

**Comment by onobc** [comment]
Hi @jardelnovaes 

Yes, this is a known issue and is a dup of #324 . The fix has been in since `2.1.0-M1` (we are at `2.1.0-M2` now.
